spec/services/email_escalated_defects_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe EmailEscalatedDefects do
+  let(:defects) { create_list(:property_defect, 3) }
+
+  describe '#call' do
+    it 'emails the team' do
+      team_message_delivery = instance_double(ActionMailer::MessageDelivery)
+      expect(EscalatedDefectsMailer).to receive(:notify)
+        .with(defects.pluck(:id))
+        .and_return(team_message_delivery)
+      expect(team_message_delivery).to receive(:deliver_later)
```

Comment:
With things like this, I've been leaning more towards testing the behaviour of the class, rather than setting up a whole bunch of stubs to test the structure of the class. Something like this should work:

```ruby
described_class.new(defects: defects)

expect(ActionMailer::DeliveryJob).to have_been_enqueued.with(EscalatedDefectsMailer, :notify, :deliver, defects.pluck(:id))
```

Reply:
I was being lazy and copied the testing from our other emails 😅 happy to write it this way 😄